### PR TITLE
Expand overlay documentation

### PR DESCRIPTION
**Pull request by TarunAdobe**
Expand overlay documentation for improving consumer experience.

## Updates

- [x] Add examples for different `overlay types`
- [x] Explain the use of a `virtual trigger` with an example
- [ ] Explain the use `placementController` in different contexts with examples 

## Related issue(s)

- #2657

## Motivation and context

Our overlay component is pretty complex, and our docs should cover those complexities. When we did a walkthrough of how to consume this component, there were a few things that weren't clear in the documentation. I think it would be great if we could write these docs in plain language and include examples for all the different usages we outline.

## How has this been tested?

-   [ ] _Test case 1_
    1. Go [here](https://chore-expand-overlay-documentation--spectrum-web-components.netlify.app/components/overlay/)
    2. Read the declarative docs
-   [ ] _Test case 2_
    1. Go [here](https://chore-expand-overlay-documentation--spectrum-web-components.netlify.app/components/imperative-api/)
    2. Read the imperative docs

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/overlay/README.md [hunk 1]
```diff
@@ -39,112 +41,17 @@ By leveraging the `trigger` attribute to pass an ID reference to another element
...
+<sp-overlay trigger="trigger@hover" placement="bottom">
+    <sp-tooltip>
+        I'm a tooltip and I'm triggered by hovering over the button!
+    </sp-tooltip>
+</sp-overlay>
 ```
 
 When a `<sp-overlay>` element is opened, it will pass that state to its direct children elements as the property `open`, which it will set to `true`. Elements should react to this by initiating any transition between closed and open that they see fit. Similarly, `open` will be set to `false` when the `<sp-overlay>` element is closed.
 
-### Action bar
```
Comment (Westbrook) [inline packages/overlay/README.md:46]: I'd like to have this somewhere. Thoughts on a good location?

File: packages/overlay/README.md [hunk 2]
```diff
@@ -168,18 +75,108 @@ When a `triggerElement` is present, either through an ID reference established v
...
+### Modal
+
+`'modal'` Overlays are opened with the `showModal()` method on a `<dialog>` element, which causes the Overlay to accept focus and trap the tab stop within the content of said Overlay.
+
+They should be used when you need to ensure that the user has interacted with the content of the Overlay before continuing with their work. This is commonly used for dialogs that require a user to confirm or cancel an action before continuing.
+
+```html
+<sp-button id="trigger">open modal</sp-button>
+<sp-overlay trigger="trigger@click" type="modal">
+    <sp-dialog-wrapper headline="Signin form" dismissable>
```
Comment (Westbrook) [inline packages/overlay/README.md:94]: We should use the `underlay` attribute here.

File: packages/overlay/README.md [hunk 3]
```diff
@@ -168,18 +75,108 @@ When a `triggerElement` is present, either through an ID reference established v
...
+</sp-overlay>
+```
+
+### Auto
+
+`'auto'` Overlays provide a place for content that is ephemeral _and_ interactive. These Overlays can accept focus but will close when losing that focus, or when interacting with other parts of the page.
+
+```html
+<sp-button id="trigger">open auto</sp-button>
+<sp-overlay trigger="trigger@click" type="auto">
```
Comment (Westbrook) [inline packages/overlay/README.md:155]: This should have a `placement` attribute.

File: packages/overlay/README.md [hunk 4]
```diff
@@ -168,18 +75,108 @@ When a `triggerElement` is present, either through an ID reference established v
...
+
+### Manual
+
+`'manual'` Overlays act much like `'auto'` Overlays, but do not close when losing focus or interacting with other parts of the page.
+
+Note: When a `'manual'` Overlay is at the top of the "overlay stack", it will still respond to the Escape key and close.
+
+```html
+<sp-button id="trigger">open manual</sp-button>
+<sp-overlay trigger="trigger@click" type="manual">
```
Comment (Westbrook) [inline packages/overlay/README.md:170]: If this doesn't accept a `placement` attribute, then the contents of the Overlay should be "full screen".

File: packages/overlay/imperative-api.md [hunk 5]
```diff
@@ -84,6 +84,135 @@ Some Overlays will always be passed focus (e.g. modal or page Overlays). When th
 
 The `trigger` option accepts an `HTMLElement` or a `VirtualTrigger` from which to position the Overlay.
 
+-   You can import the `VirtualTrigger` class from the overlay package to create a virtual trigger that can be used to position an Overlay. This is useful when you want to position an Overlay relative to a point on the screen that is not an element in the DOM, like the mouse cursor.
+
+### Using a virtual trigger
```
Comment (Westbrook) [inline packages/overlay/imperative-api.md:89]: A second or togglable version of this that showed the way that https://github.com/adobe/spectrum-web-components/issues/2568#issuecomment-1879457929 could be addressed might be nice here as well.

File: packages/overlay/imperative-api.md [hunk 6]
```diff
@@ -84,6 +84,135 @@ Some Overlays will always be passed focus (e.g. modal or page Overlays). When th
...
+
+```html-live
+<style>
+    #root {
+        position: relative;
+        width: 100%;
+        height: 20vh;
+        background-color: var(--spectrum-global-color-gray-100);
```
Comment (Westbrook) [inline packages/overlay/imperative-api.md:97]: Let's do this without `-global-` vars as they are deprecated. Maybe `--spectrum-gray-100`?

File: packages/overlay/README.md (hunk 1 above)
Comment (TarunAdobe) [reply packages/overlay/README.md:46]: If we wanna keep that then I guess the location is fine (before we talk about the API)...

File: packages/overlay/imperative-api.md (hunk 5 above)
Comment (TarunAdobe) [reply packages/overlay/imperative-api.md:89]: Should I do with an virtual trigger example or a separate simpler isolated example

File: packages/overlay/imperative-api.md (hunk 6 above)
Comment (TarunAdobe) [reply packages/overlay/imperative-api.md:97]: oops yes!!

**Review by Rajdeepc — APPROVED** [review]

**Review by Westbrook — APPROVED** [review]
Once we're caught up to `main` here, LGTM! Really appreciate you diving into the API here and getting handle on it in order to represent it to our consumers. 🙇🏼

**Pull request merged**